geth/transactions/txqueue_manager_test.go

```diff
@@ -62,21 +63,21 @@ func (s *TxQueueTestSuite) TearDownTest() {
 }
 
 func (s *TxQueueTestSuite) setupTransactionPoolAPI(account *common.SelectedExtKey, nonce hexutil.Uint64, gas hexutil.Big, txErr error) {
-	s.txServiceMock.EXPECT().GetTransactionCount(gomock.Any(), account.Address, gethrpc.PendingBlockNumber).Return(&nonce, nil)
-	s.txServiceMock.EXPECT().GasPrice(gomock.Any()).Return(big.NewInt(10), nil)
-	s.txServiceMock.EXPECT().EstimateGas(gomock.Any(), gomock.Any()).Return(&gas, nil)
-	s.txServiceMock.EXPECT().SendRawTransaction(gomock.Any(), gomock.Any()).Return(gethcommon.Hash{}, txErr)
+	s.txServiceMock.EXPECT().GetTransactionCount(gomock.Any(), account.Address, gethrpc.PendingBlockNumber).Return(&nonce, nil).AnyTimes()
+	s.txServiceMock.EXPECT().GasPrice(gomock.Any()).Return(big.NewInt(10), nil).AnyTimes()
+	s.txServiceMock.EXPECT().EstimateGas(gomock.Any(), gomock.Any()).Return(&gas, nil).AnyTimes()
+	s.txServiceMock.EXPECT().SendRawTransaction(gomock.Any(), gomock.Any()).Return(gethcommon.Hash{}, txErr).AnyTimes()
 }
 
 func (s *TxQueueTestSuite) setupStatusBackend(account *common.SelectedExtKey, password string, passwordErr error) {
 	nodeConfig, nodeErr := params.NewNodeConfig("/tmp", params.RopstenNetworkID, true)
-	s.nodeManagerMock.EXPECT().NodeConfig().Return(nodeConfig, nodeErr)
-	s.accountManagerMock.EXPECT().SelectedAccount().Return(account, nil)
+	s.nodeManagerMock.EXPECT().NodeConfig().Return(nodeConfig, nodeErr).AnyTimes()
+	s.accountManagerMock.EXPECT().SelectedAccount().Return(account, nil).AnyTimes()
 	s.accountManagerMock.EXPECT().VerifyAccountPassword(nodeConfig.KeyStoreDir, account.Address.String(), password).Return(
-		nil, passwordErr)
+		nil, passwordErr).AnyTimes()
 }
 
-func (s *TxQueueTestSuite) TestCompleteTransaction() {
+func (s *TxQueueTestSuite) setupCompleteTransaction(disableNotifications bool, args common.SendTxArgs) (*Manager, *common.QueuedTx, *common.SelectedExtKey, string) {
```

Comment:
previous approach was more readable, even though not that DRY

Reply:
I had to duplicate code for the third time so this is more readable in a different way I think.